doctor: do not throw an error if registry doesn't support ping

watilde opened the pull request:
To fix https://github.com/npm/npm/issues/15306, I've updated `npm-registry-client` at https://github.com/npm/npm-registry-client/pull/149, and updated doctor command in this PR.

Summary:
+ `./scripts/dep-update npm-registry-client 7.5.0`
+ Update ping and doctor to support getting res data directly.
+ Update npm-shrinkwrap.json

Notes:
This PR is the first step to implementing Forrest's suggest:  https://github.com/npm/npm/issues/15306#issuecomment-270287195.

watilde commented:
Removed two commits to rebase since https://github.com/npm/npm/commit/41789cffac9845603f4bdf3f5b03f412144a0e9f.

iarna commented on lib/doctor/check-ping.js:
```diff
@@ -4,9 +4,8 @@ var ping = require('../ping.js')
   var tracker = log.newItem('checkPing', 1)
   tracker.info('checkPing', 'Pinging registry')
-  ping({}, true, function (err, pong) {
-    tracker.finish()
```
I think we still should be finishing the tracker. While it doesn't matter when running `npm ping` it would help actually mark the step as completed for `npm doctor`.

watilde replied:
Oh I overlooked it! Yeah I picked it into this patch.

watilde committed:
doctor: do not throw an error

If registry doesn't support ping, npm doctor should display
the information instead of stopping its investigation.

watilde commented:
Landed in https://github.com/npm/npm/commit/9c860f2ed3bdea1417ed059b019371cd253db2ad.
